### ISSUE TITLE
[Libretro core request] Show Zapper crosshair on screen

### ISSUE THREAD

**Issue by IronYoshi64** [SourMesen/Mesen#731]
Mesen is a great standalone NES emulator and Core for RetroArch! It is awesome!

One feature that I notice that is not in the Libretro core is to always show some sort of zapper crosshair on the screen that smoothly follows your cursor/pointer movement, for the zapper/lightgun games of course. The standalone Mesen emu has this, why not the Libretro core?

I think this would be a great thing to include for the Libretro core. And for the people that may not want this, why not just have an option that is called something like "Show zapper crosshair" or something like that.

I hope this can be added sometime soon! :)

**Comment by ToniBC** [comment]
Yes, that option is missing and it is very strange because other cores do. You have to press F11 to see the Windows arrow and be able to aim with the zapper.

Hopefully they add it, in the style of Snes9x for example, that you can change the color and size of the same crosshairs.

**Comment by keithbowes** [comment]
Here on Linux, in Wayland, Mesen, FCEUmm, and Nestopia all support the Zapper.  FCEUmm and Nestopia show crosshairs, where Mesen uses the pointer's position (I like that, as for FCEUmm and Nestopia, I have to use mouse grabbing to hide the pointer so that I pay attention to the crosshair instead of the pointer).

However, in the much faster KMS (where there's no mouse pointer; there is the GPM daemon to allow copying and pasting, but that's not available when EGL is being used to render things), the situation is:
* FCEUmm: works perfectly.
* Nestopia: a crosshair is visible but the mouse buttons don't shoot.
* Mesen: no crosshair (or pointer) but the mouse buttons shoot (so if a crosshair were correctly set up, it would probably work as well or better than in FCEUmm).

Judging from the source of FCEUmm and Nestopia, what's required is to draw a crosshair via the libretro API and set up a callback to change its position when the  pointer-device position changes (the easy part), and report the change back to the core (i.e. whatever Mesen class::method() is used to set the Zapper's coordinates on the screen).

**Comment by ToniBC** [comment]
Although in Mesen it works as you comment, it is the mouse pointer itself and you have to press F11 to capture the movement. Other cores, as you say, have that implemented.

Aiming a crosshair or a red square is also easier than using the mouse cursor.

Applying it should be simple and the same method of other cores should work without problems.
